src/DataStructures.jl

```diff
@@ -16,7 +16,9 @@ module DataStructures
                  eachindex, keytype, valtype
     import Base: iterate
 
-    using InteractiveUtils: methodswith
+    using OrderedCollections
+    using OrderedCollections: not_iterator_of_pairs, rehash!
+    import OrderedCollections: filter, filter!, isordered
```

Comment:
`filter` and `filter!` both come from `Base`
Why are they being imported?

Reply:
Well, they have to be imported because DataStructures extends them. I imported them from OrderedCollections instead of Base due to the `Iterators.filter` separation---I basically thought "this should be the same `filter` that OrderedCollections uses, whatever version of `filter` that is."